.gitlab/cancel-prev-pipelines.yml

```diff
@@ -0,0 +1,23 @@
+cancel-prev-pipelines:
+  stage: .pre
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
+  tags: ["arch:amd64"]
+  rules: # this should only run on dev branches
+  - if: $CI_PIPELINE_SOURCE =~ /^schedule.*$/
+    when: never
+  - if: $CI_COMMIT_TAG
+    when: never
+  - if: $CI_COMMIT_REF_NAME == "main"
+    when: never
+  - if: $CI_COMMIT_REF_NAME =~ /^[6-7]\.[0-9]+.*/
+    when: never
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏ Worth future proofing for Agent 8.0?

Reply:
We could, but is agent 8 even planned?

Reply:
Agreed, this snippet is already used in `.gitlab-ci.yml` if you need:
```
.if_release_branch: &if_release_branch
  if: $CI_COMMIT_BRANCH =~ /^[0-9]+\.[0-9]+\.x$/
```